### PR TITLE
Some cleanup.

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Removed code used only for testing & debugging purposes.

### CODE REVIEW

**Comment by ExtremeFiretop** [comment]
Latest result :)

image

May send off a release soon.

**Comment by Martinski4GitHub** [comment]
> Latest result :)
> 
> 
Looking good. BTW,  do you know if you had any 3rd-party cron job entries in crontab when MerlinAU was executed?
I'm asking because I don't see any messages about cron jobs in your log after stopping Entware services.

**Comment by ExtremeFiretop** [comment]
> > Latest result :)
> 
> Looking good. BTW, do you know if you had any 3rd-party cron job entries in crontab when MerlinAU was executed? I'm asking because I don't see any messages about cron jobs in your log after stopping Entware services.

I for sure have cron jobs. I have things like spdmerlin and Backupmon and diversion and unbound all installed and scheduled with cron jobs configured for stuff. I don't think any of them were being executed at the time.

I was only looking at the fact that it supressed the eject USB message.

**Comment by Martinski4GitHub** [comment]
> > > Latest result :)
> > 
> > 
> > Looking good. BTW, do you know if you had any 3rd-party cron job entries in crontab when MerlinAU was executed? I'm asking because I don't see any messages about cron jobs in your log after stopping Entware services.
> 
> I for sure have cron jobs. I have things like spdmerlin and Backupmon and diversion and unbound all installed and scheduled with cron jobs configured for stuff. I don't think any of them were being executed at the time.
>
>
If you have those cron job entries then there should be log messages about removing them. Something is odd here. Can you send me a screenshot of your current list of scheduled cron jobs?

**Comment by ExtremeFiretop** [comment]
Screenshot of: **cru l**

image

Copy of the MerlinAU **log file** for that flash:

```
2024-10-02 04:43:11 MerlinAU v1.3.1
2024-10-02 04:43:11 Running the update task now... Checking for F/W updates...
2024-10-02 04:43:13 Required RAM: 101814 KB - RAM Free: 329836 KB - RAM Available: 1125728 KB
2024-10-02 04:43:13 Backup Started (by BACKUPMON)
2024-10-02 04:44:02 Backup Finished
2024-10-02 04:44:02 Backup Completed Successfully
2024-10-02 04:44:02 Required RAM: 101814 KB - RAM Free: 326480 KB - RAM Available: 1122204 KB
2024-10-02 04:44:02 -----------------------------------------------------------
2024-10-02 04:44:02 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
2024-10-02 04:44:02   Length      Date    Time    Name
2024-10-02 04:44:03 ---------  ---------- -----   ----
2024-10-02 04:44:03  74547428  09-26-2024 00:58   GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
2024-10-02 04:44:03      9691  09-26-2024 00:36   README-merlin.txt
2024-10-02 04:44:03      5698  09-26-2024 00:58   Changelog-3006.txt
2024-10-02 04:44:03       128  09-26-2024 00:58   sha256sum.sha256
2024-10-02 04:44:03  --------                     -------
2024-10-02 04:44:03  74562945                     4 files
2024-10-02 04:44:03 -----------------------------------------------------------
2024-10-02 04:44:05 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
2024-10-02 04:44:05   inflating: GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
2024-10-02 04:44:05   inflating: Changelog-3006.txt
2024-10-02 04:44:05   inflating: sha256sum.sha256
2024-10-02 04:44:05 -----------------------------------------------------------
2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253412 KB - RAM Available: 1049088 KB
2024-10-02 04:44:05 Current version NOT found in changelog file. Bypassing changelog verification for this run.
2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253656 KB - RAM Available: 1049332 KB
2024-10-02 04:44:05 SHA256 signature check for firmware image file passed successfully.
2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253672 KB - RAM Available: 1049596 KB
2024-10-02 04:44:05 Router Web URL is: http://192.168.50.1
2024-10-02 04:44:46 The email notification was sent successfully [START_FW_UPDATE_STATUS].
2024-10-02 04:44:46 Stopping Diversion service...
2024-10-02 04:44:51 Flashing GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb... Please wait for reboot in about 4 minutes or less.
2024-10-02 04:44:53 Searching for Entware services to stop...
2024-10-02 04:44:53 Stopping Entware services...
2024-10-02 04:44:53 -----------------------------------------------------------
2024-10-02 04:44:53 /opt/etc/init.d/S61unbound
2024-10-02 04:44:53 -----------------------------------------------------------
```

Contents of the crontab file located here: /var/spool/cron/crontabs

image

```
45 */6 * * * /jffs/addons/amtm/routerdate cron #amtm_RouterDate#
10 7 * * Sun /bin/sh /jffs/addons/amtm/sc_update.mod -run #amtm_ScriptsUpdateNotification#
1 7 * * * /bin/sh /jffs/addons/amtm/led_control.mod -on #amtm_LEDcontrol_on#
41 18 * * * /bin/sh /jffs/addons/amtm/led_control.mod -off #amtm_LEDcontrol_off#
15 3 * * Sat,Tue /bin/sh /jffs/addons/amtm/led_control.mod -upd #amtm_LEDcontrol_update#
0 4 * * * sh /jffs/scripts/MerlinAU.sh run_now #MerlinAU#
30 3 * * * sh /jffs/scripts/backupmon.sh #RunBackupMon#
12 4 * * * curl -o \/opt\/var\/lib\/unbound\/root\.hints https://www.internic.net/domain/named.cache #root_servers#
*/15 * * * * /jffs/addons/unbound/unbound_rpz.sh download #Unbound_RPZ.sh#
00 2 * * Sun /bin/sh /opt/share/diversion/file/update-bl.div reset #Diversion_UpdateBL#
20 5 * * * /bin/sh /opt/share/diversion/file/rotate-logs.div #Diversion_RotateLogs#
20 17 * * * diversion count_ads count #Diversion_CountAds#
0 */1 * * * /jffs/scripts/spdmerlin generate #spdMerlin#
*/5 * * * * service restart_letsencrypt #LetsEncrypt#
```

**Comment by Martinski4GitHub** [comment]
> Screenshot of: **cru l**
> 
> image
> 
> Copy of the MerlinAU **log file** for that flash:
> 
> ```
> 2024-10-02 04:43:11 MerlinAU v1.3.1
> 2024-10-02 04:43:11 Running the update task now... Checking for F/W updates...
> 2024-10-02 04:43:13 Required RAM: 101814 KB - RAM Free: 329836 KB - RAM Available: 1125728 KB
> 2024-10-02 04:43:13 Backup Started (by BACKUPMON)
> 2024-10-02 04:44:02 Backup Finished
> 2024-10-02 04:44:02 Backup Completed Successfully
> 2024-10-02 04:44:02 Required RAM: 101814 KB - RAM Free: 326480 KB - RAM Available: 1122204 KB
> 2024-10-02 04:44:02 -----------------------------------------------------------
> 2024-10-02 04:44:02 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
> 2024-10-02 04:44:02   Length      Date    Time    Name
> 2024-10-02 04:44:03 ---------  ---------- -----   ----
> 2024-10-02 04:44:03  74547428  09-26-2024 00:58   GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
> 2024-10-02 04:44:03      9691  09-26-2024 00:36   README-merlin.txt
> 2024-10-02 04:44:03      5698  09-26-2024 00:58   Changelog-3006.txt
> 2024-10-02 04:44:03       128  09-26-2024 00:58   sha256sum.sha256
> 2024-10-02 04:44:03  --------                     -------
> 2024-10-02 04:44:03  74562945                     4 files
> 2024-10-02 04:44:03 -----------------------------------------------------------
> 2024-10-02 04:44:05 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
> 2024-10-02 04:44:05   inflating: GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
> 2024-10-02 04:44:05   inflating: Changelog-3006.txt
> 2024-10-02 04:44:05   inflating: sha256sum.sha256
> 2024-10-02 04:44:05 -----------------------------------------------------------
> 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253412 KB - RAM Available: 1049088 KB
> 2024-10-02 04:44:05 Current version NOT found in changelog file. Bypassing changelog verification for this run.
> 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253656 KB - RAM Available: 1049332 KB
> 2024-10-02 04:44:05 SHA256 signature check for firmware image file passed successfully.
> 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253672 KB - RAM Available: 1049596 KB
> 2024-10-02 04:44:05 Router Web URL is: http://192.168.50.1
> 2024-10-02 04:44:46 The email notification was sent successfully [START_FW_UPDATE_STATUS].
> 2024-10-02 04:44:46 Stopping Diversion service...
> 2024-10-02 04:44:51 Flashing GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb... Please wait for reboot in about 4 minutes or less.
> 2024-10-02 04:44:53 Searching for Entware services to stop...
> 2024-10-02 04:44:53 Stopping Entware services...
> 2024-10-02 04:44:53 -----------------------------------------------------------
> 2024-10-02 04:44:53 /opt/etc/init.d/S61unbound
> 2024-10-02 04:44:53 -----------------------------------------------------------
> ```
> 
> Contents of the crontab file located here: /var/spool/cron/crontabs
> 
> image
> 
> ```
> 45 */6 * * * /jffs/addons/amtm/routerdate cron #amtm_RouterDate#
> 10 7 * * Sun /bin/sh /jffs/addons/amtm/sc_update.mod -run #amtm_ScriptsUpdateNotification#
> 1 7 * * * /bin/sh /jffs/addons/amtm/led_control.mod -on #amtm_LEDcontrol_on#
> 41 18 * * * /bin/sh /jffs/addons/amtm/led_control.mod -off #amtm_LEDcontrol_off#
> 15 3 * * Sat,Tue /bin/sh /jffs/addons/amtm/led_control.mod -upd #amtm_LEDcontrol_update#
> 0 4 * * * sh /jffs/scripts/MerlinAU.sh run_now #MerlinAU#
> 30 3 * * * sh /jffs/scripts/backupmon.sh #RunBackupMon#
> 12 4 * * * curl -o \/opt\/var\/lib\/unbound\/root\.hints https://www.internic.net/domain/named.cache #root_servers#
> */15 * * * * /jffs/addons/unbound/unbound_rpz.sh download #Unbound_RPZ.sh#
> 00 2 * * Sun /bin/sh /opt/share/diversion/file/update-bl.div reset #Diversion_UpdateBL#
> 20 5 * * * /bin/sh /opt/share/diversion/file/rotate-logs.div #Diversion_RotateLogs#
> 20 17 * * * diversion count_ads count #Diversion_CountAds#
> 0 */1 * * * /jffs/scripts/spdmerlin generate #spdMerlin#
> */5 * * * * service restart_letsencrypt #LetsEncrypt#
> ```

Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow.
Have a good night, bud.

**Comment by ExtremeFiretop** [comment]
> > Screenshot of: **cru l**
> > image
> > Copy of the MerlinAU **log file** for that flash:
> > ```
> > 2024-10-02 04:43:11 MerlinAU v1.3.1
> > 2024-10-02 04:43:11 Running the update task now... Checking for F/W updates...
> > 2024-10-02 04:43:13 Required RAM: 101814 KB - RAM Free: 329836 KB - RAM Available: 1125728 KB
> > 2024-10-02 04:43:13 Backup Started (by BACKUPMON)
> > 2024-10-02 04:44:02 Backup Finished
> > 2024-10-02 04:44:02 Backup Completed Successfully
> > 2024-10-02 04:44:02 Required RAM: 101814 KB - RAM Free: 326480 KB - RAM Available: 1122204 KB
> > 2024-10-02 04:44:02 -----------------------------------------------------------
> > 2024-10-02 04:44:02 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
> > 2024-10-02 04:44:02   Length      Date    Time    Name
> > 2024-10-02 04:44:03 ---------  ---------- -----   ----
> > 2024-10-02 04:44:03  74547428  09-26-2024 00:58   GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
> > 2024-10-02 04:44:03      9691  09-26-2024 00:36   README-merlin.txt
> > 2024-10-02 04:44:03      5698  09-26-2024 00:58   Changelog-3006.txt
> > 2024-10-02 04:44:03       128  09-26-2024 00:58   sha256sum.sha256
> > 2024-10-02 04:44:03  --------                     -------
> > 2024-10-02 04:44:03  74562945                     4 files
> > 2024-10-02 04:44:03 -----------------------------------------------------------
> > 2024-10-02 04:44:05 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-BE98_PRO_firmware/GT-BE98_PRO_firmware.zip
> > 2024-10-02 04:44:05   inflating: GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb
> > 2024-10-02 04:44:05   inflating: Changelog-3006.txt
> > 2024-10-02 04:44:05   inflating: sha256sum.sha256
> > 2024-10-02 04:44:05 -----------------------------------------------------------
> > 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253412 KB - RAM Available: 1049088 KB
> > 2024-10-02 04:44:05 Current version NOT found in changelog file. Bypassing changelog verification for this run.
> > 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253656 KB - RAM Available: 1049332 KB
> > 2024-10-02 04:44:05 SHA256 signature check for firmware image file passed successfully.
> > 2024-10-02 04:44:05 Required RAM: 101814 KB - RAM Free: 253672 KB - RAM Available: 1049596 KB
> > 2024-10-02 04:44:05 Router Web URL is: http://192.168.50.1
> > 2024-10-02 04:44:46 The email notification was sent successfully [START_FW_UPDATE_STATUS].
> > 2024-10-02 04:44:46 Stopping Diversion service...
> > 2024-10-02 04:44:51 Flashing GT-BE98_PRO_3006_102.2_alpha1-g6d259892ed_nand_squashfs.pkgtb... Please wait for reboot in about 4 minutes or less.
> > 2024-10-02 04:44:53 Searching for Entware services to stop...
> > 2024-10-02 04:44:53 Stopping Entware services...
> > 2024-10-02 04:44:53 -----------------------------------------------------------
> > 2024-10-02 04:44:53 /opt/etc/init.d/S61unbound
> > 2024-10-02 04:44:53 -----------------------------------------------------------
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Contents of the crontab file located here: /var/spool/cron/crontabs
> > image
> > ```
> > 45 */6 * * * /jffs/addons/amtm/routerdate cron #amtm_RouterDate#
> > 10 7 * * Sun /bin/sh /jffs/addons/amtm/sc_update.mod -run #amtm_ScriptsUpdateNotification#
> > 1 7 * * * /bin/sh /jffs/addons/amtm/led_control.mod -on #amtm_LEDcontrol_on#
> > 41 18 * * * /bin/sh /jffs/addons/amtm/led_control.mod -off #amtm_LEDcontrol_off#
> > 15 3 * * Sat,Tue /bin/sh /jffs/addons/amtm/led_control.mod -upd #amtm_LEDcontrol_update#
> > 0 4 * * * sh /jffs/scripts/MerlinAU.sh run_now #MerlinAU#
> > 30 3 * * * sh /jffs/scripts/backupmon.sh #RunBackupMon#
> > 12 4 * * * curl -o \/opt\/var\/lib\/unbound\/root\.hints https://www.internic.net/domain/named.cache #root_servers#
> > */15 * * * * /jffs/addons/unbound/unbound_rpz.sh download #Unbound_RPZ.sh#
> > 00 2 * * Sun /bin/sh /opt/share/diversion/file/update-bl.div reset #Diversion_UpdateBL#
> > 20 5 * * * /bin/sh /opt/share/diversion/file/rotate-logs.div #Diversion_RotateLogs#
> > 20 17 * * * diversion count_ads count #Diversion_CountAds#
> > 0 */1 * * * /jffs/scripts/spdmerlin generate #spdMerlin#
> > */5 * * * * service restart_letsencrypt #LetsEncrypt#
> > ```
> 
> Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow. Have a good night, bud.

Chat more tomorrow.

I'm realizing now that when I first tested I tested on my test AXE11000 with a single cron job so I didn't disturb my regular internet.

But this run above was me flashing the latest alpha on the 3006 router. I'm expecting that's making a difference.

**Comment by Martinski4GitHub** [comment]
> > > Screenshot of: **cru l**
>> > ...
> > > Copy of the MerlinAU **log file** for that flash:
> > >  ...
> > >   
> > > Contents of the crontab file located here: /var/spool/cron/crontabs
>>> ...
> > > 45 */6 * * * /jffs/addons/amtm/routerdate cron #amtm_RouterDate#
> > > 10 7 * * Sun /bin/sh /jffs/addons/amtm/sc_update.mod -run #amtm_ScriptsUpdateNotification#
> > > 1 7 * * * /bin/sh /jffs/addons/amtm/led_control.mod -on #amtm_LEDcontrol_on#
> > > 41 18 * * * /bin/sh /jffs/addons/amtm/led_control.mod -off #amtm_LEDcontrol_off#
> > > 15 3 * * Sat,Tue /bin/sh /jffs/addons/amtm/led_control.mod -upd #amtm_LEDcontrol_update#
> > > 0 4 * * * sh /jffs/scripts/MerlinAU.sh run_now #MerlinAU#
> > > 30 3 * * * sh /jffs/scripts/backupmon.sh #RunBackupMon#
> > > 12 4 * * * curl -o \/opt\/var\/lib\/unbound\/root\.hints https://www.internic.net/domain/named.cache #root_servers#
> > > */15 * * * * /jffs/addons/unbound/unbound_rpz.sh download #Unbound_RPZ.sh#
> > > 00 2 * * Sun /bin/sh /opt/share/diversion/file/update-bl.div reset #Diversion_UpdateBL#
> > > 20 5 * * * /bin/sh /opt/share/diversion/file/rotate-logs.div #Diversion_RotateLogs#
> > > 20 17 * * * diversion count_ads count #Diversion_CountAds#
> > > 0 */1 * * * /jffs/scripts/spdmerlin generate #spdMerlin#
> > > */5 * * * * service restart_letsencrypt #LetsEncrypt#
> > > ```
> > 
> > 
> > Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow. Have a good night, bud.
> 
> Chat more tomorrow. Goodnight buddy.
> 
> I'm realizing now that when I first tested I tested on my test AXE11000 with a single cron job so I didn't disturb my regular internet.
> 
> But this run above was me flashing the latest alpha on the 3006 router. I'm expecting that's making a difference.
>
>
Hmmm. I'm expecting the **3006.102.x** F/W to behave the same WRT cron jobs, crontab, & cru command....
You have a lot of 3rd-party cron jobs but for some reason, there are no log messages at all about them. I'm puzzled.
I'll have to sleep on this... :>)

**Comment by ExtremeFiretop** [comment]
> > > > Screenshot of: **cru l**
> > > > ...
> > > > Copy of the MerlinAU **log file** for that flash:
> > > > ...
> > > > Contents of the crontab file located here: /var/spool/cron/crontabs
> > > > ...
> > > > 45 */6 * * * /jffs/addons/amtm/routerdate cron #amtm_RouterDate#
> > > > 10 7 * * Sun /bin/sh /jffs/addons/amtm/sc_update.mod -run #amtm_ScriptsUpdateNotification#
> > > > 1 7 * * * /bin/sh /jffs/addons/amtm/led_control.mod -on #amtm_LEDcontrol_on#
> > > > 41 18 * * * /bin/sh /jffs/addons/amtm/led_control.mod -off #amtm_LEDcontrol_off#
> > > > 15 3 * * Sat,Tue /bin/sh /jffs/addons/amtm/led_control.mod -upd #amtm_LEDcontrol_update#
> > > > 0 4 * * * sh /jffs/scripts/MerlinAU.sh run_now #MerlinAU#
> > > > 30 3 * * * sh /jffs/scripts/backupmon.sh #RunBackupMon#
> > > > 12 4 * * * curl -o /opt/var/lib/unbound/root.hints https://www.internic.net/domain/named.cache #root_servers#
> > > > */15 * * * * /jffs/addons/unbound/unbound_rpz.sh download #Unbound_RPZ.sh#
> > > > 00 2 * * Sun /bin/sh /opt/share/diversion/file/update-bl.div reset #Diversion_UpdateBL#
> > > > 20 5 * * * /bin/sh /opt/share/diversion/file/rotate-logs.div #Diversion_RotateLogs#
> > > > 20 17 * * * diversion count_ads count #Diversion_CountAds#
> > > > 0 */1 * * * /jffs/scripts/spdmerlin generate #spdMerlin#
> > > > */5 * * * * service restart_letsencrypt #LetsEncrypt#
> > > 
> > > 
> > > Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow. Have a good night, bud.
> > 
> > 
> > Chat more tomorrow. Goodnight buddy.
> > I'm realizing now that when I first tested I tested on my test AXE11000 with a single cron job so I didn't disturb my regular internet.
> > But this run above was me flashing the latest alpha on the 3006 router. I'm expecting that's making a difference.
> 
> Hmmm. I'm expecting the **3006.102.x** F/W to behave the same WRT cron jobs, crontab, & cru command.... You have a lot of 3rd-party cron jobs but for some reason, there are no log messages at all about them. I'm puzzled. I'll have to sleep on this... :>)

Now you got me curious...
In the code I see it should say "Cron jobs from 3rd-party add-ons were not found." which I didn't get.

Looking at the script on the router I can confirm that the function _RemoveCronJobsFromAddOns_ is there. Let me try again just for fun, the reason is I noticed in my log file for the MerlinAU run, it was at 4:44 AM, but my MerlinAU script has a modified date of  8AM.

I'm wondering if it's possible my primary router just had an old copy of the dev version, maybe I never copied it to my primary after testing it on my AXE11000, and only once I made the changes to up the minimum version did I get the _RemoveCronJobsFromAddOns_  function.

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

This looks more promising ;)

image

I likely had "1.3.1" with only the changelog changes, before your PR. Then updated it once I made my changes for my PR again lol.

**Comment by Martinski4GitHub** [comment]
> > > > 
> > > > 
> > > > Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow. Have a good night, bud.
> > > 
> > > 
> > > Chat more tomorrow. Goodnight buddy.
> > > I'm realizing now that when I first tested I tested on my test AXE11000 with a single cron job so I didn't disturb my regular internet.
> > > But this run above was me flashing the latest alpha on the 3006 router. I'm expecting that's making a difference.
> > 
> > 
> > Hmmm. I'm expecting the **3006.102.x** F/W to behave the same WRT cron jobs, crontab, & cru command.... You have a lot of 3rd-party cron jobs but for some reason, there are no log messages at all about them. I'm puzzled. I'll have to sleep on this... :>)
> 
> Now you got me curious... In the code I see it should say "Cron jobs from 3rd-party add-ons were not found." which I didn't get.
> 
> Looking at the script on the router I can confirm that the function _RemoveCronJobsFromAddOns_ is there. Let me try again just for fun, the reason is I noticed in my log file for the MerlinAU run, it was at 4:44 AM, but my MerlinAU script has a modified date of 8AM.
> 
> I'm wondering if it's possible my primary router just had an old copy of the dev version, maybe I never copied it to my primary after testing it on my AXE11000, and only once I made the changes to up the minimum version did I update the script and get the _RemoveCronJobsFromAddOns_ function.
>
>
Good catch about the timestamp!! LOL!!
I hope that is what happened because I've also reviewed the function and I can't see anything obvious that would prevent the code from seeing all the cron jobs you have and logging some messages about them.

**Comment by ExtremeFiretop** [comment]
> > > > > Great, thanks. Please don't stay up any longer only for this. There's no urgency. This can wait until tomorrow. Have a good night, bud.
> > > > 
> > > > 
> > > > Chat more tomorrow. Goodnight buddy.
> > > > I'm realizing now that when I first tested I tested on my test AXE11000 with a single cron job so I didn't disturb my regular internet.
> > > > But this run above was me flashing the latest alpha on the 3006 router. I'm expecting that's making a difference.
> > > 
> > > 
> > > Hmmm. I'm expecting the **3006.102.x** F/W to behave the same WRT cron jobs, crontab, & cru command.... You have a lot of 3rd-party cron jobs but for some reason, there are no log messages at all about them. I'm puzzled. I'll have to sleep on this... :>)
> > 
> > 
> > Now you got me curious... In the code I see it should say "Cron jobs from 3rd-party add-ons were not found." which I didn't get.
> > Looking at the script on the router I can confirm that the function _RemoveCronJobsFromAddOns_ is there. Let me try again just for fun, the reason is I noticed in my log file for the MerlinAU run, it was at 4:44 AM, but my MerlinAU script has a modified date of 8AM.
> > I'm wondering if it's possible my primary router just had an old copy of the dev version, maybe I never copied it to my primary after testing it on my AXE11000, and only once I made the changes to up the minimum version did I update the script and get the _RemoveCronJobsFromAddOns_ function.
> 
> Good catch about the timestamp!! LOL!! I hope that is what happened because I've also reviewed the function and I can't see anything obvious that would prevent the code from seeing all the cron jobs you have and logging some messages about them.

I've had my head out of the game for a month, distracted with other things; clearly it works fine and I was just running "1.3.1" but I just had an outdated version lol. 

Checking the timeline of the modified time was my only hint, nothing in the code seemed weird to me. 
I spent a 5 minutes testing it for 3006 just now, ran it line by line, everything returned results as expected. :D

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> This looks more promising ;)
> ...
> 
YES!!! Those are the log messages I would expect based on your cron jobs list. So, the function is working as designed on the newer F/W version as well. 

>
>
> I likely had "1.3.1" with only the changelog changes, before your PR. Then updated it once I made my changes for my PR again lol.
>
Thanks for staying up to run the script & verify the "real" latest code. Now we both can go to sleep more relaxed. Have a great night, bud. Sleep tight :>)
